File: app/src/main/java/com/lessonscontrol/data/dao/LessonDAO.java
package com.lessonscontrol.data.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.lessonscontrol.data.entities.Lesson;

import java.util.List;

/**
 * DAO for {@link Lesson} objects.
 *
 * @author hblonski
 */
@Dao
public interface LessonDAO {

    @Insert
    void insert(Lesson lesson);

    @Delete
    void delete(Lesson lesson);

    @Update
    void update(Lesson lesson);

    @Query("SELECT * FROM lesson WHERE les_student = :studentID ORDER BY les_nextdate ASC")
    LiveData<List<Lesson>> findLessonsByStudent(long studentID);
}

File: app/src/main/java/com/lessonscontrol/data/dao/LessonDAO.kt
package com.lessonscontrol.data.dao

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*
import com.lessonscontrol.data.entities.Lesson

/**
 * DAO for [Lesson] objects.
 *
 * @author hblonski
 */
@Dao
interface LessonDAO {

    @Insert
    fun insert(lesson: Lesson)

    @Delete
    fun delete(lesson: Lesson)

    @Update
    fun update(lesson: Lesson)

    @Query("SELECT * FROM lesson WHERE les_student = :studentID ORDER BY les_nextdate ASC")
    fun findLessonsByStudent(studentID: Long): LiveData<List<Lesson>>
}
